### PR TITLE
Proxy authenticate implement

### PR DESCRIPTION
**Pull request by luckyboy125**

### CODE REVIEW

File: lib/src/proxy-authenticate.class.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import { BaseHeader } from "./base-header.class";
+export class ProxyAuthenticate extends BaseHeader {
+  protected _type?: string | undefined;
+  protected _realm?: string | undefined;
+
+  getType(): string | undefined {
+    return this._type;
+  }
+
+  setType(value: string | undefined) {
```
Comment (mwilkins91) [inline lib/src/proxy-authenticate.class.ts:10]: Can we add more specific types here? They are listed on this page: https://developer.mozilla.org/en-US/docs/Web/HTTP/Authentication#authentication_schemes

Can probably make a "[Authentication schemes](https://developer.mozilla.org/en-US/docs/Web/HTTP/Authentication#authentication_schemes)" type

```typescript
type AuthenticationType = 'Basic' | 'Bearer' | etc...
```

```suggestion
  setType(value: AuthenticationType | undefined) {
```

File: lib/src/proxy-authenticate.class.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import { BaseHeader } from "./base-header.class";
+export class ProxyAuthenticate extends BaseHeader {
+  protected _type?: string | undefined;
+  protected _realm?: string | undefined;
```
Comment (mwilkins91) [inline lib/src/proxy-authenticate.class.ts:4]:
```suggestion
  /** 
    * @description A description of the protected area, the realm. If no realm is 
    * specified, clients often display a formatted host name instead.
    * protected _realm?: string | undefined;
    */
```

**Review by mwilkins91 — CHANGES_REQUESTED** [review]
Couple suggestions for more specific types & comments

File: lib/src/proxy-authenticate.class.ts (hunk 1 above)
Comment (luckyboy125) [reply lib/src/proxy-authenticate.class.ts:10]: Okay I will check and modify as specific types

File: lib/src/proxy-authenticate.class.ts (hunk 2 above)
Comment (luckyboy125) [reply lib/src/proxy-authenticate.class.ts:4]: Should I use description for every headers?

**Pull request merged**